task/common/machine/storage.go

```diff
@@ -130,6 +132,8 @@ func Transfer(ctx context.Context, source, destination string, include string) e
 		return err
 	}
 
+	defer progress(2 * time.Second)()
```

Comment:
> _alternatively, can we inject \r instead of \n into logs? Would be able to keep progress on one line 👼_

[Nice try](https://xkcd.com/565). No it's not possible, because Terraform already adds a `\n` for us at the end of every `logrus.Info` call. If you want to go down that path, you can always `tput smcup`, `tput rmcup` and `tput cup` to hijack the whole terminal. 🤢

Reply:
> _strongly suggest something like ${TF_LOG_TPI_SYNC_INTERVAL:-10}_ 
> [...]
> _user configurable disowns responsibility_

That's a great milestone in our cardinal sin roadmap: using environment variables to alter the behavior of a provider. I'd mildly suggest to resist that temptation, especially when customizing this setting requires an awkward, hard to document action from users.

Alternatively, what if we bump this to `10*time.Second` to reduce the noise?

> _2sec -> 10sec means ~500 messages/GB -> ~100 messages/GB (on an 8Mbps internet connection)_

What does messages/Gb mean? 🤔

Reply:
* #479 or unresolve 🧤 ⚔️

Reply:
> using environment variables

seems to be the approach in https://www.terraform.io/plugin/log/managing. Anyway missing the point - it's fine if there's another way (e.g. a variable in `main.tf`) the point is to be user-configurable?

Reply:
> _seems to be the approach in https://www.terraform.io/plugin/log/managing_

Indeed. Even those variables (that are officially supported and documented) aren't meant for users, but for [provider/core] developers. Parsing additional variables from providers (instead of Terraform Core) is still further from the desired user experience. 

See also https://github.com/iterative/terraform-provider-iterative/pull/423#pullrequestreview-901140497

Reply:
Moved to https://github.com/iterative/terraform-provider-iterative/pull/479#discussion_r839965657